src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs

```diff
@@ -57,9 +88,54 @@ public Result<AssessmentEvent> SelectSuitableAssessmentEvent(IAssessmentEventSel
                     KnowledgeComponentId = KnowledgeComponent.Id,
                     AssessmentEventId = result.Value.Id
                 });
+
             return result;
         }
 
+        private void TryPass()
+        {
+            if (IsPassed)
+                return;
+
+            if (Mastery >= 0.95)
+            {
+                Causes(new KnowledgeComponentPassed()
+                {
+                    KnowledgeComponentId = KnowledgeComponent.Id,
+                    LearnerId = LearnerId
+                });
+                TrySatisfy();
+            }
+        }
+
+        private void TryComplete(bool isCompletedBeforeSubmission)
+        {
+            if (isCompletedBeforeSubmission)
```

Comment:
Why is it important to check if `isCompletedBeforeSubmission` when we check IsCompleted in the next line?